2022-08-01 07:38:32,905 INFO     ---------------------------------------------------------------------------------------------
2022-08-01 07:38:32,906 INFO     Geo: cqd
2022-08-01 07:38:32,906 INFO     Data Path: data/FB15k-237-betae
2022-08-01 07:38:32,906 INFO     #entity: 14505
2022-08-01 07:38:32,906 INFO     #relation: 474
2022-08-01 07:38:32,906 INFO     #max steps: 150000
2022-08-01 07:38:32,906 INFO     Evaluate unoins using: DNF
2022-08-01 07:38:32,906 INFO     loading data
2022-08-01 07:39:14,727 INFO     Training info:
2022-08-01 07:39:14,739 INFO     2p: 149689
2022-08-01 07:39:17,180 INFO     Validation info:
2022-08-01 07:39:17,181 INFO     2p: 5000
2022-08-01 07:39:17,184 INFO     Test info:
2022-08-01 07:39:17,184 INFO     2p: 5000
2022-08-01 07:39:19,850 INFO     Model Parameter Configuration:
2022-08-01 07:39:19,850 INFO     Parameter embeddings.0.weight: torch.Size([14505, 2000]), require_grad = True
2022-08-01 07:39:19,850 INFO     Parameter embeddings.1.weight: torch.Size([474, 2000]), require_grad = True
2022-08-01 07:39:19,850 INFO     Parameter score_transform_fun.weight: torch.Size([2, 2000]), require_grad = True
2022-08-01 07:39:19,850 INFO     Parameter score_transform_fun.bias: torch.Size([2]), require_grad = True
2022-08-01 07:39:19,851 INFO     Parameter score_transform_ent_fun.weight: torch.Size([2, 2000]), require_grad = True
2022-08-01 07:39:19,851 INFO     Parameter score_transform_ent_fun.bias: torch.Size([2]), require_grad = True
2022-08-01 07:39:19,851 INFO     Parameter Number: 29966004
2022-08-01 07:39:19,870 INFO     Loading checkpoint models/fb15k-237-betae...
2022-08-01 07:39:20,275 INFO     tasks = 2p
2022-08-01 07:39:20,275 INFO     init_step = 99999
2022-08-01 07:39:20,275 INFO     Start Training...
2022-08-01 07:39:20,275 INFO     learning_rate = 0
2022-08-01 07:39:20,275 INFO     batch_size = 1024
2022-08-01 07:39:20,275 INFO     hidden_dim = 1000
2022-08-01 07:39:20,275 INFO     gamma = 12.000000
/home/shinaliu/miniconda3/envs/kg/lib/python3.8/site-packages/torch/autograd/__init__.py:173: UserWarning: Error detected in DivBackward0. Traceback of forward call that caused the error:
  File "main.py", line 632, in <module>
    main(parse_args())
  File "main.py", line 537, in main
    log = KGReasoning.train_step(model, optimizer, train_path_iterator, args, step)
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/models.py", line 674, in train_step
    model.forward(positive_sample=positive_sample,
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/base.py", line 439, in forward
    scores = d2.query_2p(entity_embeddings=self.embeddings[0],
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/discrete.py", line 103, in query_2p
    res = t_norm(atom1_scores_3d, atom2_scores_3d)
  File "/home/shinaliu/miniconda3/envs/kg/lib/python3.8/site-packages/torchnorms/tnorms/classic.py", line 63, in __call__
    return (a * b) / (1.0 + (1.0 - a) * (1.0 - b))
 (Triggered internally at  ../torch/csrc/autograd/python_anomaly_mode.cpp:104.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
posix.uname_result(sysname='Linux', nodename='gonzo-605-3.local', release='3.10.0-1160.49.1.el7.x86_64', version='#1 SMP Tue Nov 30 15:51:32 UTC 2021', machine='x86_64')
Traceback (most recent call last):
  File "main.py", line 632, in <module>
    main(parse_args())
  File "main.py", line 537, in main
    log = KGReasoning.train_step(model, optimizer, train_path_iterator, args, step)
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/models.py", line 931, in train_step
    loss.backward()
  File "/home/shinaliu/miniconda3/envs/kg/lib/python3.8/site-packages/torch/_tensor.py", line 363, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/shinaliu/miniconda3/envs/kg/lib/python3.8/site-packages/torch/autograd/__init__.py", line 173, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: CUDA out of memory. Tried to allocate 908.00 MiB (GPU 0; 10.92 GiB total capacity; 9.80 GiB already allocated; 401.38 MiB free; 9.85 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
